### PR TITLE
implemented homework

### PR DESCRIPTION
**Pull request by mmmmmcr**
Varianta 1.

### CODE REVIEW

**Comment by mmmmmcr** [comment]
check first variant

File: data/src/main/java/com/example/data/repo/NewsRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Scheduler;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+
+public class NewsRepositoryImpl implements NewsRepository {
+    RemoteSource source;
```
Comment (Misca) [inline data/src/main/java/com/example/data/repo/NewsRepositoryImpl.java]: we could pe making these fields private and final, also let's have same naming (remoteDataStore, localDataStore )

File: data/src/main/java/com/example/data/repo/remote/RemoteSource.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.example.data.repo.remote.NewsAPI;
+
+import java.util.Locale;
+
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+
+public class RemoteSource {
+    private static final String API_KEY = "7da6b88ba5a7463994eb22961ea43aa0";
+    NewsAPI newsAPI;
```
Comment (Misca) [inline data/src/main/java/com/example/data/repo/remote/RemoteSource.java]: let's hane the newsAPI final and private

File: data/src/main/java/com/example/data/repo/remote/RemoteSource.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.example.data.dtos.ArticleListDto;
+import com.example.data.repo.remote.NewsAPI;
+
+import java.util.Locale;
+
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+
+public class RemoteSource {
```
Comment (Misca) [inline data/src/main/java/com/example/data/repo/remote/RemoteSource.java]: let's rename this class, our local class is called NewsLocalDataStore, for consistency let's rename this one to NewsRemoteDataStore (naming conventions are part of an architecture)

**Review by Misca — CHANGES_REQUESTED** [review]
almost there 👍

**Review by Misca — APPROVED** [review]
GG 👍

**Pull request merged**